cloud_functions/flagCryptominer.js

```diff
@@ -0,0 +1,35 @@
+const utils = require('./utils')
+const { ValidationError }  = require('./ValidationError')
+
+
+const run = (mixpanel, message) => {
+  try {
+    const userSubjectId = validate(message)
+    if (userSubjectId) {
+      mixpanel.people.set(`google:${userSubjectId}`, { 'isCryptominer': true }, err => { if (err) throw err })
+    }
+  } catch (error) {
+    console.error(utils.coerceToError(error))
+  }
+}
+
+const validate = message => {
+  if (!(message.attributes && message.attributes.cryptominer === true)) {
+    throw new ValidationError('Unexpected message; missing `cryptominer: true` attribute. Is the Pub/Sub subscription set up correctly?')
+  }
+  let data
+  try {
+    data = message.data && JSON.parse(Buffer.from(message.data, 'base64').toString())
+  } catch (error) {
+    throw new ValidationError(error.message)
+  }
```

Comment:
ToL: Must turn of my JVM spectacles. Rethrowing errors like this in the JVM would lose stack context, but from my first-google-hit this may be normal in JS land... https://stackoverflow.com/questions/25763837/how-to-specify-a-caused-by-in-a-javascript-error

Reply:
Indeed. I struggled with this myself.

As @panentheos pointed out, `ValidationError` may not even be needed. I had originally thought to use it to help isolate per-invocation errors (validation) from per-instance (i.e. Cloud Function instance, specifically fetching the Mixpanel token and initializing the Mixpanel SDK). However, that looks like it turned out to not be necessary so I'll probably remove it.

A happy consequence of that is that this `try/catch` will no longer be necessary, which also means `data` can be `const`. No more `let`!